gradle build should enforce minimal jvm version

Its very unintuitive today: this happens often on accident because i switch JAVA_HOME to java 7 when backporting (and sometimes forget to switch it back). 

But in general, users may not be aware java 8 is required. Currently exceptions look like this:
```
:buildSrc:build UP-TO-DATE
=======================================
Elasticsearch Build Hamster says Hello!
=======================================
  Gradle Version : 2.8
  JDK Version    : 1.7.0_55-b13 (Oracle Corporation)
  OS Info        : Linux 4.2.0-16-generic (amd64)
:core:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':core:compileJava'.
> invalid flag: -Xdoclint:all/private

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -59,6 +59,11 @@ class BuildPlugin implements Plugin<Project> {
                 throw new GradleException('Gradle 2.6 or above is required to build elasticsearch')
             }
 
+            // enforce Java version
+            if (JavaVersion.current().compareTo(JavaVersion.VERSION_1_8) < 0) {
+                throw new GradleException('Java 8 or above is required to build Elasticsearch')
+            }
+
             // Build debugging info
             println '======================================='
             println 'Elasticsearch Build Hamster says Hello!'

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -60,7 +60,7 @@ class BuildPlugin implements Plugin<Project> {
             }
 
             // enforce Java version
-            if (JavaVersion.current().compareTo(JavaVersion.VERSION_1_8) < 0) {
+            if (!JavaVersion.current().isJava8Compatible()) {
                 throw new GradleException('Java 8 or above is required to build Elasticsearch')
             }
 

